components/butterfly/src/rumor/service_file.rs

```diff
@@ -76,27 +76,20 @@ impl ServiceFile {
                       body }
     }
 
-    /// Encrypt the contents of the service file
-    pub fn encrypt(&mut self, user_pair: &BoxKeyPair, service_pair: &BoxKeyPair) -> Result<()> {
-        self.body = user_pair.encrypt(&self.body, Some(service_pair))?
-                             .into_bytes();
-        self.encrypted = true;
-        Ok(())
-    }
-
     /// Return the body of the service file as a stream of bytes. Always returns a new copy, due to
     /// the fact that we might be encrypted.
-    pub fn body(&self, cache_key_path: &Path) -> Result<Vec<u8>> {
```

Comment:
This was a brain-fart on my part... For some reason I was using a `Cow` internally, rather than just returning the vector of bytes :facepalm: 

It would be interesting to actually return the `Cow`, but with how the other types are, it looks like it ends up complicating a number of other types and signatures, so that doesn't seem useful at the moment. Oh well :shrug: 

I'll clean up the code and remove the `Cow`s.